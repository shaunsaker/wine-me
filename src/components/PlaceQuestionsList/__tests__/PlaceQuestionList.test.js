import React from 'react';
import renderer from 'react-test-renderer';

import PlaceQuestionList from '..';

jest.mock('../PlaceQuestion', () => 'PlaceQuestion');

describe('PlaceQuestionList', () => {
  const spies = [];
  const dispatch = jest.fn();
  const data = [{ id: 'foo' }, { id: 'bar' }];

  describe('renders', () => {
    it('renders with minimum required props', () => {
      const component = renderer.create(<PlaceQuestionList />);

      expect(component).toMatchSnapshot();
    });

    it('renders the default state', () => {
      const component = renderer.create(<PlaceQuestionList data={data} slideIndex={0} />);

      expect(component).toMatchSnapshot();
    });
  });

  describe('methods', () => {
    it('should handle scrollToIndex', () => {
      const component = renderer.create(<PlaceQuestionList data={data} slideIndex={0} />);
      const instance = component.getInstance();

      instance.scrollToIndex(1);
    });
  });

  describe('lifecycle methods', () => {
    it('should call scrollToIndex in componentDidUpdate if slideIndex changed', () => {
      spies[0] = jest.spyOn(PlaceQuestionList.prototype, 'scrollToIndex');
      const component = renderer.create(<PlaceQuestionList data={data} slideIndex={0} />);
      const nextSlideIndex = 1;

      component.update(<PlaceQuestionList data={data} slideIndex={nextSlideIndex} />);

      expect(spies[0]).toHaveBeenCalledWith(nextSlideIndex);
    });
  });

  afterEach(() => {
    spies.forEach((spy) => {
      if (spy) {
        spy.mockClear();
      }
    });
    dispatch.mockClear();
  });
});
