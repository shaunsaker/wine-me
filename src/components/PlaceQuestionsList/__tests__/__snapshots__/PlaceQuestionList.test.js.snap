// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlaceQuestionList renders renders the default state 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  data={
    Array [
      Object {
        "id": "foo",
      },
      Object {
        "id": "bar",
      },
    ]
  }
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  pagingEnabled={true}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "width": 690,
            },
          ]
        }
      >
        <PlaceQuestion
          category={
            Object {
              "id": "foo",
            }
          }
          handleSetValue={[Function]}
        />
      </View>
    </View>
    <View
      style={
        Array [
          Object {
            "flexDirection": "row",
          },
          null,
        ]
      }
    >
      <View
        style={
          Array [
            Object {
              "justifyContent": "center",
            },
            Object {
              "width": 690,
            },
          ]
        }
      >
        <PlaceQuestion
          category={
            Object {
              "id": "bar",
            }
          }
          handleSetValue={[Function]}
        />
      </View>
    </View>
  </View>
</RCTScrollView>
`;

exports[`PlaceQuestionList renders renders with minimum required props 1`] = `
<RCTScrollView
  contentContainerStyle={Object {}}
  disableVirtualization={false}
  getItem={[Function]}
  getItemCount={[Function]}
  getItemLayout={[Function]}
  horizontal={true}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onContentSizeChange={[Function]}
  onEndReachedThreshold={2}
  onLayout={[Function]}
  onMomentumScrollEnd={[Function]}
  onScroll={[Function]}
  onScrollBeginDrag={[Function]}
  onScrollEndDrag={[Function]}
  pagingEnabled={true}
  renderItem={[Function]}
  scrollEnabled={false}
  scrollEventThrottle={50}
  stickyHeaderIndices={Array []}
  style={Object {}}
  updateCellsBatchingPeriod={50}
  viewabilityConfigCallbackPairs={Array []}
  windowSize={21}
>
  <View />
</RCTScrollView>
`;
